### PR TITLE
fix:  calling survey countly events - WPB-11629

### PR DESCRIPTION
**Pull request by netbe**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-11629" title="WPB-11629" target="_blank">BugWPB-11629</a>  Rework call survey events
  </td></table>
  <br />
 

### Issue

Since we reworked the setup for analytics, the calling survey was not functional and produced a crash when accessing deprecated implementation `Analytics.shared`.

**Solution**

* remove all old code about Analytics and AnalyticsType
* move existing logic to AnalyticsTrackingEvent
* add usecase to wrap the CallSurveyEvent invocation to expose it to Wire-iOS

### Testing

1. Initiate call on A
2. Answer on B and answer form

Screenshot 2024-10-16 at 22 32 45

Screenshot 2024-10-16 at 22 53 27

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

**Review by agisilaos — COMMENTED** [review]
Looking good! Thank you @netbe!

**Review by agisilaos — APPROVED** [review]

File: WireAnalytics/Sources/WireAnalytics/Events/AnalyticsEvent+Predefined.swift [hunk 1]
```diff
@@ -100,4 +102,54 @@ public extension AnalyticsEvent {
...
+    /// - Returns: A callQualitySurvey analytics event.
+
+    static func callQualitySurvey(_ review: CallQualitySurveyReview) -> AnalyticsEvent {
+        .init(name: "calling.call_quality_review", segmentation: review.segmentation)
+    }
+}
+
+/// User review for call quality.
+
+public enum CallQualitySurveyReview {
```
Comment (johnxnguyen) [inline WireAnalytics/Sources/WireAnalytics/Events/AnalyticsEvent+Predefined.swift]: Suggestion: perhaps move to it's own file.

File: WireAnalytics/Sources/WireAnalytics/Events/Segmentation/SegmentationEntry+Predefined.swift [hunk 2]
```diff
@@ -83,4 +85,37 @@ extension SegmentationEntry {
         .init(key: "conversation_size", value: value.analyticsValue)
     }
 
+    /// Creates a `SegmentationEntry` for the score of the calling survey
+    ///
+    /// - Parameter value: The score from 1 to 5.
+    /// - Returns: A `SegmentationEntry` instance with the appropriate key and value.
+
+    static func score(_ value: Int) -> Self {
```
Comment (johnxnguyen) [inline WireAnalytics/Sources/WireAnalytics/Events/Segmentation/SegmentationEntry+Predefined.swift]:
```suggestion
    static func callScore(_ value: Int) -> Self {
```

File: WireAnalytics/Sources/WireAnalytics/Events/Segmentation/SegmentationEntry+Predefined.swift [hunk 3]
```diff
@@ -83,4 +85,37 @@ extension SegmentationEntry {
...
+        .init(key: "score", value: String(value))
+    }
+
+    /// Creates a `SegmentationEntry` for the duration of the calling survey
+    ///
+    /// - Parameter value: The duration of the call.
+    /// - Returns: A `SegmentationEntry` instance with the appropriate key and value.
+
+    static func duration(_ value: TimeInterval) -> Self {
```
Comment (johnxnguyen) [inline WireAnalytics/Sources/WireAnalytics/Events/Segmentation/SegmentationEntry+Predefined.swift]:
```suggestion
    static func callDuration(_ value: TimeInterval) -> Self {
```

File: WireAnalytics/Sources/WireAnalytics/Events/Segmentation/SegmentationEntry+Predefined.swift [hunk 4]
```diff
@@ -83,4 +85,37 @@ extension SegmentationEntry {
...
+
+    static func callLabel(_ value: String) -> Self {
+        .init(key: "label", value: value)
+    }
+
+    /// Creates a `SegmentationEntry` for the ignore reason of the calling survey
+    /// - Parameter value: the ignore reason
+    /// - Returns: A `SegmentationEntry` instance with the appropriate key and value.
+
+    static func ignoreReason(_ value: String) -> Self {
```
Comment (johnxnguyen) [inline WireAnalytics/Sources/WireAnalytics/Events/Segmentation/SegmentationEntry+Predefined.swift]:
```suggestion
    static func callIgnoreReason(_ value: String) -> Self {
```

File: wire-ios-sync-engine/Source/Use cases/CallQualitySurveyUseCase.swift [hunk 5]
```diff
@@ -16,13 +16,23 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
-import WireUtilities
+import WireAnalytics
 
-extension CoreDataStack {
+/// sourcery: AutoMockable
+public protocol CallQualitySurveyUseCaseProtocol {
```
Comment (johnxnguyen) [inline wire-ios-sync-engine/Source/Use cases/CallQualitySurveyUseCase.swift]:
```suggestion
public protocol SubmitCallQualitySurveyUseCaseProtocol {
```

**Review by johnxnguyen — APPROVED** [review]

**Pull request merged**